blockchain/blockchain.go

```diff
@@ -281,25 +285,22 @@ func (b *Blockchain) verifyBlock(txn db.Transaction, block *core.Block) error {
 		return err
 	}
 
+	expectedBlockNumber := uint64(0)
+	expectedParentHash := &felt.Zero
+
 	head, err := b.head(txn)
 	if err != nil && !errors.Is(err, db.ErrKeyNotFound) {
```

Comment:
Maybe we could make it a little bit shorter like:
```
if err == nil {
    expectedBlockNumber = head.Number + 1
    expectedParentHash = head.Hash
} else if !errors.Is(err, db.ErrKeyNotFound) {
    return err
}
```

Reply:
yeap, looks neater. done